[quant][graphmode][fx] Remove inplace option for convert_fx

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46955 [quant][graphmode][fx] Remove inplace option for convert_fx**

Summary:
originally `inplace` option is used in eager mode to indicate that we want to swap float submodules to quantized submodules directly in the input model. This makes sense in eager mode since it does not change code. however in fx graph mode we generate new GraphModule out of the input nn.Module and we can't inplace construct a GraphModule given an nn.Module, therefore inplace option does not make sense in fx passes.

Initially we were thinking of adding a `invalidate_quantized_float_parameters` option to free the memory
of quantized floating parameters, but it turns out we will do module swap just like in eager mode for the modules
that are quantized, so the old floating point module will not be referenced after quantization. therefore this feature
is only needed for functionals, since most people are using quantization with modules we may not need this.

we'll revisit after we find there is a need for this.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24579400](https://our.internmc.facebook.com/intern/diff/D24579400)

jerryzh168 committed:
[quant][graphmode][fx] Remove inplace option for convert_fx

Summary:
Initially we were thinking of adding a `invalidate_quantized_float_parameters` option to free the memory
of quantized floating parameters, but it turns out we will do module swap just like in eager mode for the modules
that are quantized, so the old floating point module will not be referenced after quantization. therefore this feature
is only needed for functionals, since most people are using quantization with modules we may not need this.

we'll revisit after we find there is a need for this.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Remove inplace option for convert_fx"


Summary:
Initially we were thinking of adding a `invalidate_quantized_float_parameters` option to free the memory
of quantized floating parameters, but it turns out we will do module swap just like in eager mode for the modules
that are quantized, so the old floating point module will not be referenced after quantization. therefore this feature
is only needed for functionals, since most people are using quantization with modules we may not need this.

we'll revisit after we find there is a need for this.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24579400](https://our.internmc.facebook.com/intern/diff/D24579400)

[ghstack-poisoned]

supriyar reviewed (approved):
For sake of posterity, can we also update the description to include why we cannot have an inplace option for fx?

jerryzh168 committed:
Update on "[quant][graphmode][fx] Remove inplace option for convert_fx"


Summary:
Initially we were thinking of adding a `invalidate_quantized_float_parameters` option to free the memory
of quantized floating parameters, but it turns out we will do module swap just like in eager mode for the modules
that are quantized, so the old floating point module will not be referenced after quantization. therefore this feature
is only needed for functionals, since most people are using quantization with modules we may not need this.

we'll revisit after we find there is a need for this.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24579400](https://our.internmc.facebook.com/intern/diff/D24579400)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Remove inplace option for convert_fx"


Summary:
originally `inplace` option is used in eager mode to indicate that we want to swap float submodules to quantized submodules directly in the input model. This makes sense in eager mode since it does not change code. however in fx graph mode we generate new GraphModule out of the input nn.Module and we can't inplace construct a GraphModule given an nn.Module, therefore inplace option does not make sense in fx passes.

Initially we were thinking of adding a `invalidate_quantized_float_parameters` option to free the memory
of quantized floating parameters, but it turns out we will do module swap just like in eager mode for the modules
that are quantized, so the old floating point module will not be referenced after quantization. therefore this feature
is only needed for functionals, since most people are using quantization with modules we may not need this.

we'll revisit after we find there is a need for this.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24579400](https://our.internmc.facebook.com/intern/diff/D24579400)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Remove inplace option for convert_fx"


Summary:
originally `inplace` option is used in eager mode to indicate that we want to swap float submodules to quantized submodules directly in the input model. This makes sense in eager mode since it does not change code. however in fx graph mode we generate new GraphModule out of the input nn.Module and we can't inplace construct a GraphModule given an nn.Module, therefore inplace option does not make sense in fx passes.

Initially we were thinking of adding a `invalidate_quantized_float_parameters` option to free the memory
of quantized floating parameters, but it turns out we will do module swap just like in eager mode for the modules
that are quantized, so the old floating point module will not be referenced after quantization. therefore this feature
is only needed for functionals, since most people are using quantization with modules we may not need this.

we'll revisit after we find there is a need for this.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24579400](https://our.internmc.facebook.com/intern/diff/D24579400)

[ghstack-poisoned]
